### PR TITLE
Exo24

### PR DESCRIPTION
**Pull request by tianyuchen**

### CODE REVIEW

File: 024-Lexicographic_permutations.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+def factorial(n):
+    if n == 1:
+        return n
+    else:
+        return n * factorial(n - 1)
+
+factorial(len(digits))   # 10! = 3628800
+factorial(len(digits) - 1)   # 9! = 362880, ex. when first digit is zero, there are 9! permutations
+0
```
[the remark this answers is not in the document]
Comment (tianyuchen) [reply 024-Lexicographic_permutations.py]: Fixed

File: 024-Lexicographic_permutations.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+# 2 * 9! < 1000000 (th lexicographic permutation) < 3 * 9! so the first number is 2
+n = 9
+objective = 10 ** 6 - 1
+answer = ""
+
+while n > 0:
+    tmp = divmod(objective, factorial(n))
+    # the next digit is the round number of objective divide by n!
```
[the remark this answers is not in the document]
Comment (tianyuchen) [reply 024-Lexicographic_permutations.py]: Fixed

File: 024-Lexicographic_permutations.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+factorial(len(digits))   # 10! = 3628800
+factorial(len(digits) - 1)   # 9! = 362880, ex. when first digit is zero, there are 9! permutations
+0
+# 2 * 9! < 1000000 (th lexicographic permutation) < 3 * 9! so the first number is 2
+n = 9
+objective = 10 ** 6 - 1
+answer = ""
+
+while n > 0:
+    tmp = divmod(objective, factorial(n))
```
[the remark this answers is not in the document]
Comment (tianyuchen) [reply 024-Lexicographic_permutations.py]: Fixed

File: 024-Lexicographic_permutations.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        return n
+    else:
+        return n * factorial(n - 1)
+
+factorial(len(digits))   # 10! = 3628800
+factorial(len(digits) - 1)   # 9! = 362880, ex. when first digit is zero, there are 9! permutations
+0
+# 2 * 9! < 1000000 (th lexicographic permutation) < 3 * 9! so the first number is 2
+n = 9
+objective = 10 ** 6 - 1
```
[the remark this answers is not in the document]
Comment (tianyuchen) [reply 024-Lexicographic_permutations.py]: I would like to keep this as objective.

[the remark this answers is not in the document]
Comment (gameboy1024) [reply 024-Lexicographic_permutations.py]: You don't need this variable. It's equivalent and interchangable with remainder at all times. Check line 28 and line 32.

[the remark this answers is not in the document]
Comment (tianyuchen) [reply 024-Lexicographic_permutations.py]: eh.. I know, but use objective explains better

**Pull request merged**